src/Raven.Server/Documents/Indexes/IndexFieldsPersistence.cs

```diff
@@ -92,6 +96,32 @@ internal bool TryReadNumberOfDimensions(string fieldName, out int dimensions)
             return _vectorFieldsDimensions.TryGetValue(fieldName, out dimensions);
         }
 
+        internal bool TryReadEmbeddingsGenerationTaskIdentifier(string fieldName, out string taskName)
+        {
+            return _embeddingsGenerationTaskIdentifiers.TryGetValue(fieldName, out taskName);
+        }
+
+        internal void SetEmbeddingsGenerationTaskIdentifier(string fieldName, string taskIdentifier)
+        {
+            var isStoredOnDisk = _embeddingsGenerationTaskIdentifiers.TryGetValue(fieldName, out var diskStoredSourceEtlTaskName);
+
+            PortableExceptions.ThrowIf<InvalidOperationException>(isStoredOnDisk && diskStoredSourceEtlTaskName != taskIdentifier, $"We are expecting that field {fieldName} has the same ETL task as it was before. However, stored ETL task was {diskStoredSourceEtlTaskName} and current one is {taskIdentifier}.");
+
+            var isStoredInRuntime = false;
+            if (_embeddingsGenerationTaskIdentifiersToWrite != null)
+            {
+                isStoredInRuntime = _embeddingsGenerationTaskIdentifiersToWrite.TryGetValue(fieldName, out var runtimeStoredSourceEtlTaskName);
+                PortableExceptions.ThrowIf<InvalidOperationException>(isStoredInRuntime && runtimeStoredSourceEtlTaskName != taskIdentifier,
+                    $"We are expecting that field {fieldName} has the same ETL task as it was before. However, previously ETL task was {runtimeStoredSourceEtlTaskName} and current one is {taskIdentifier}.");
```

Comment:
We should replace "ETL task" by "Embeddings Generation task" in the exceptions messages